tests/test_smartspim.py

```diff
@@ -0,0 +1,272 @@
+"""
+Tests the SmartSPIM acquisition metadata creation
+"""
+
+import copy
+import os
+import shutil
+import tempfile
+import unittest
+from unittest.mock import MagicMock, patch
+
+from aind_data_schema.core import acquisition
+
+from aind_metadata_mapper.smartspim.acquisition import (
+    JobSettings,
+    SmartspimETL,
+)
+
+from .resources.smartspim.example_metadata import (
+    example_filter_mapping,
+    example_metadata_info,
+    example_processing_manifest,
+    example_session_end_time,
+)
+
+
+class TestSmartspimETL(unittest.TestCase):
+    """Tests methods in the SmartSPIM class"""
+
+    def setUp(self):
+        """Setting up temporary folder directory"""
+        self.temp_dir = tempfile.mkdtemp()
+        self.example_job_settings_success = JobSettings(
+            subject_id="000000",
+            raw_dataset_path="/SmartSPIM_000000_2024-10-10_10-10-10",
+            output_directory=self.temp_dir,
+            asi_filename="derivatives/ASI_logging.txt",
+            mdata_filename_json="derivatives/metadata.json",
+            processing_manifest_path="derivatives/processing_manifest.json",
+        )
+        self.example_smartspim_etl_success = SmartspimETL(
+            job_settings=self.example_job_settings_success
+        )
+
+        self.example_job_settings_fail_mouseid = JobSettings(
+            subject_id="00000",
+            raw_dataset_path="/SmartSPIM_00000_2024-10-10_10-10-10",
+            output_directory=self.temp_dir,
+            asi_filename="derivatives/ASI_logging.txt",
+            mdata_filename_json="derivatives/metadata.json",
+            processing_manifest_path="derivatives/processing_manifest.json",
+        )
+        self.example_smartspim_etl_fail_mouseid = SmartspimETL(
+            job_settings=self.example_job_settings_fail_mouseid
+        )
+
+    def tearDown(self):
+        """Tearing down temporary folder directory"""
+        if os.path.exists(self.temp_dir):
```

Comment:
Same thing here. It's always a bit risky modifying a user's directory. We can try to mock the write operation and just verify the contents of the call are what's expected